### ISSUE TITLE
Can't underline in cterm mode 

### ISSUE THREAD

**Issue by juliandkatz** [joshdick/onedark.vim#198]
- [ x ] I have carefully read and followed the [Installation section of the README](https://github.com/joshdick/onedark.vim#installation)
- [ x ] I have searched through the [closed issues](https://github.com/joshdick/onedark.vim/issues?q=is%3Aissue+is%3Aclosed) to see if there is an existing solution for my issue

### Terminal Emulator Info

```
alacritty 0.3.3 (0815774)
tmux 2.9a
```

### Output From `vim --version`

```
NVIM v0.4.3
```

### Issue Description

I'm unable to set my `Search` results to be underlined, as they were in my previous font.

![no-underline-search](https://user-images.githubusercontent.com/8396513/72489245-565ace80-37c8-11ea-8e14-25463a1cab7e.png)

I've customized a couple other things, like brightening the Normal text and making the search results orange.  Below this configuration, you'll see that I've tried setting a variety of ways to underline the Search results.

```
Plug 'joshdick/onedark.vim'

  augroup colorextend
    autocmd!

    " Make the main text lighter
    autocmd ColorScheme * call onedark#extend_highlight("Normal", { "fg": { "cterm": "253" } })

    " Make search results orange
    autocmd ColorScheme * call onedark#extend_highlight("Search", { "fg": { "cterm": "214" }})

    " Remove search result background
    autocmd ColorScheme * call onedark#extend_highlight("Search", { "bg": { "cterm": "NONE" }})

    " Make search results underlined
    autocmd ColorScheme * call onedark#extend_highlight("Search", { "cterm": "underline" })
    autocmd ColorScheme * call onedark#extend_highlight("Search", { "gui": "underline" })
    autocmd ColorScheme * call onedark#extend_highlight("Search", { "term": "underline" })
  augroup END

  syntax on
```

I found another issue where this ended up being a problem with someone's font.  To confirm that this was not the case, I tried setting the highlight manually using the following vim command:

```
:hi Search cterm=underline
```

This resulted in underlined search results:

![underlined-search-results](https://user-images.githubusercontent.com/8396513/72489422-d41eda00-37c8-11ea-9554-e8437d895a20.png)

Thanks for your time!  Please let me know if I've missed a detail or should try something different.  Otherwise, loving the font.  The floating window coloring is especially good, and appreciating the customizability.

Cheers!

**Comment by joshdick** [comment]
Try this, making sure the snippet appears **before** `colorscheme onedark` in your configuration:

```vim
augroup colorextend
  autocmd!

  " Make the main text lighter
  autocmd ColorScheme * call onedark#extend_highlight("Normal", { "fg": { "cterm": "253" } })

  " Make search results underlined
  let s:yellow = { "gui": "#E5C07B", "cterm": "180", "cterm16": "3" }
  autocmd ColorScheme * call onedark#set_highlight("Search", { "fg": s:yellow, "gui": "underline", "cterm": "underline" })
augroup END

```

You can also use a similar technique to override the `IncSearch` styling if you wish.

**Comment by juliandkatz** [comment]
Works!  Thanks for figuring it out.

For curiosity's sake, why didn't mine work?  Does extend_highlight only allow for one extension of a certain highlight?

**Comment by joshdick** [comment]
@juliandkatz 

No problem, you're welcome!

Yours could have been made to work depending on how you have both `onedark.vim` and Vim itself configured. For example, if you have `set termguicolors` enabled in your Vim configuration, the following line from your original snippet would have no effect since you'd have to reference`gui` and not `ceterm`:

```
autocmd ColorScheme * call onedark#extend_highlight("Search", { "bg": { "cterm": "NONE" }})
```

The solution I posted is more direct and succinct since it overrides all possible configuration options for a specific highlight group at once, including removing the background by not explicitly specifying one.

I hope that makes sense!

I'm going to close this issue since we've resolved it, but feel free to keep the conversation going!
